app/browser/webtorrent.js

```diff
@@ -43,6 +52,64 @@ function send (msg) {
   channel.send(messages.TORRENT_MESSAGE, msg)
 }
 
+function setupFiltering () {
+  Filtering.registerHeadersReceivedFilteringCB(function (details, isPrivate) {
+    if (details.method !== 'GET') {
+      return {}
+    }
+    if (!isTorrentFile(details)) {
+      return {}
+    }
+
+    var viewerUrl = getViewerURL(details.url)
+
+    return {
+      responseHeaders: {
+        'Location': [ viewerUrl ]
+      },
+      statusLine: 'HTTP/1.1 301 Moved Permanently',
+      resourceName: 'webtorrent'
+    }
+  })
+}
+
+/**
+ * Check if the request is a torrent file.
+ * @param {Object} details First argument of the webRequest.onHeadersReceived
+ *                         event. The properties "responseHeaders" and "url"
+ *                         are read.
+ * @return {boolean} True if the resource is a torrent file.
+ */
+function isTorrentFile (details) {
```

Comment:
This .torrent file detection is pretty conservative. It looks for a torrent mimetype, and only uses the ".torrent" extension to indicate a torrent if the server sets the 'content-type' to 'application/octet-stream'.